specifications/bcastByz/bcastByz.tla

```diff
@@ -0,0 +1,983 @@
+ * with InitNoBcast, we can rewrite UnforgLtl as a first-order formula.    
+ *)          
+Unforg == (\A i \in Proc: i \in Corr => (pc[i] /= "AC")) 
+
+
+(* If a correct process broadcasts, then every correct process accepts.     *)
+CorrLtl == (\A i \in Corr: pc[i] = "V1") => <>(\E i \in Corr: pc[i] = "AC")
+
+(* If a correct process accepts, then every other correct process accepts.  *)
+RelayLtl == []((\E i \in Corr: pc[i] = "AC") => <>(\A i \in Corr: pc[i] = "AC"))
+
+(* It is the indutive invariant for the property Unforgeability which is used
+ * by TLC to generate initial states. This inductive invariant is equivalent 
+ * to IndInv_Unforg. The reasons why we use two inductive are
+ *  1/ I started with TypeOK and tried to add new constraints in order to have
+ *     an inductive invariant. I think it is the nature way to find an inductive
+ *     invariant. Moreover, writing a TLAPS proof for IndInv_Unforg is easy 
+ *     to organize and decompose.
+ *  2/ However, checking IndInv_Unforg with TLC is inefficient because TLC
+ *     generates too many unnecessary initial states. When TLC evaluates the
+ *     expression pc \in [ Proc -> {"V0", "V1", "SE", "AC"} ], it will generate
+ *     all possible states. Too many. IndInv_Unforg_TLC helps TLC reduce the
+ *     number of initial states.    
+ *)                
+IndInv_Unforg_TLC ==  
+  /\ pc = [ i \in Proc |-> "V0" ]
+  /\ Corr \in SUBSET Proc
+  /\ Cardinality( Corr ) >= N - T
+  /\ Faulty = Proc \ Corr
+  /\ \A i \in Proc : pc[i] /= "AC"
+  /\ sent = {}  
+  /\ rcvd \in [ Proc -> sent \cup SUBSET ByzMsgs ]
+  
+IndInv_Unforg ==  
+  /\ TypeOK
+  /\ FCConstraints
+  /\ sent = {}  
+  /\ pc = [ i \in Proc |-> "V0" ]        
+ 
+(* If the user wants to check IndInv_Unforg_TLC, he should add a counter to know 
+ * how many transition steps are considered. It must be at most 1.
+ * Notice that: we don't care about newMessages in IndInv_Unforg(_TLC).
+ *) 
+Init0 == IndInv_Unforg_TLC /\ newMessages = {}                                          
+
+
+Next0 ==  
+  /\ (\E self \in Corr: Step(self))
+(* remember that only moving forward one step
+ * when checking an inductive invariant with TLC
+ */\ nStep = 1 
+ *)
+ 
+Spec0 == Init0 /\ [][Next0]_vars 
+               /\ WF_vars(\E self \in Corr: /\ Receive(self)
+                                            /\ \/ UponV1(self)
+                                               \/ UponNonFaulty(self)
+                                               \/ UponAcceptNotSent(self)
+                                               \/ UponAccept(self))
+
+(* Some theories about cardinalities used in our proof but cannot proved with TLAPS
+ * CardThm, FiniteSetThm1 are copied from Lamport's handout, for more info please
+ * visit http://research.microsoft.com/en-us/um/people/lamport/handout.pdf. 
+ * CardThm2, SetEnumProp, PWUCard, FiniteSetThm2, UMCardThm, ByzMsgsCard, and ByzMsgsProp 
+ * are added by Thanh Hai.  
+ *)
```

Comment:
Please check out the standard module FiniteSetTheorems that comes with TLAPS: it contains most, if not all, of your unproved lemmas. In fact, it is not true that these cannot be proved using TLAPS, although their proofs are not automatic. If you want to check the proofs, have a look at module FiniteSetTheorems_proofs.